src/main/java/com/minecolonies/coremod/colony/ColonyView.java

```diff
@@ -39,79 +39,71 @@
-    private       String                              name        = "Unknown";
-    private int      dimensionId;
-    private BlockPos center = BlockPos.ORIGIN;
-
+    private final Map<Integer, CitizenDataView>       citizens           = new HashMap<>();
     /**
-     * Defines if workers are hired manually or automatically.
+     * Last barbarian spawnpoints.
      */
-    private boolean manualHiring = false;
-
+    private final List<BlockPos>                      lastSpawnPoints    = new ArrayList<>();
     /**
-     * Defines if workers are housed manually or automatically.
+     * The Positions which players can freely interact.
      */
-    private boolean manualHousing = false;
-
-    //  Buildings
-    @Nullable
-    private BuildingTownHall.View townHall;
-    private int maxCitizens = 0;
-
+    private final Set<BlockPos>                       freePositions      = new HashSet<>();
     /**
-     * Check if the colony has a warehouse.
+     * The Blocks which players can freely interact with.
      */
-    private boolean hasColonyWarehouse;
-
+    private final Set<Block>                          freeBlocks         = new HashSet<>();
     /**
-     * Last barbarian spawnpoints.
+     * The Set of waypoints.
      */
-    private final List<BlockPos> lastSpawnPoints = new ArrayList<>();
-
+    private final Set<BlockPos>                       wayPoints          = new HashSet<>();
     /**
-     * The Positions which players can freely interact.
+     * Datas about the happiness of a colony
      */
-    private final Set<BlockPos> freePositions = new HashSet<>();
-
+    private final HappinessData                       happinessData      = new HappinessData();
+    private       String                              name               = "Unknown";
+    private       int                                 dimensionId;
+    private       BlockPos                            center             = BlockPos.ORIGIN;
     /**
-     * The Blocks which players can freely interact with.
+     * Defines if workers are hired manually or automatically.
      */
-    private final Set<Block> freeBlocks = new HashSet<>();
-
+    private       boolean                             manualHiring       = false;
     /**
-     * The Set of waypoints.
+     * Defines if workers are housed manually or automatically.
      */
-    private final Set<BlockPos> wayPoints = new HashSet<>();
-
+    private       boolean                             manualHousing      = false;
+    //  Buildings
+    @Nullable
+    private       BuildingTownHall.View               townHall;
+    private       int                                 maxCitizens        = 0;
+    /**
+     * Check if the colony has a warehouse.
+     */
+    private       boolean                             hasColonyWarehouse;
     /**
      * The overall happiness of the colony.
      */
-    private double overallHappiness = 5;
-
+    private       double                              overallHappiness   = 5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)